lightning/src/routing/network_graph.rs

```diff
@@ -374,18 +374,26 @@ impl<C: Deref , L: Deref > RoutingMessageHandler for NetGraphMsgHandler<C, L> wh
 		let mut pending_events = self.pending_events.lock().unwrap();
 		let batch_count = batches.len();
 		for (batch_index, batch) in batches.into_iter().enumerate() {
-			// Per spec, the initial first_blocknum needs to be <= the query's first_blocknum and subsequent
-			// must be >= the prior reply. We'll simplify this by using zero since its still spec compliant and
-			// sequence completion is now explicitly.
-			let first_blocknum = 0;
-
-			// Per spec, the final end_blocknum needs to be >= the query's end_blocknum, so we'll use the
-			// query's value. Prior batches must use the number of blocks that fit into the message. We'll
-			// base this off the last SCID in the batch since we've somewhat abusing first_blocknum.
+			// Per spec, the initial `first_blocknum` needs to be <= the query's `first_blocknum`
+			// and subsequent `first_blocknum`s must be >= the prior reply's `first_blocknum`.
+			// We'll simplify this by using the query's first_blocknum since its still spec
+			// compliant and sequence completion is now explicit.
+			let first_blocknum = msg.first_blocknum;
+
+			// Per spec, the last end_block needs to be >= the query's end_blocknum.
+			// Each message carries the number of blocks its contents fit in, though there is no
+			// requirement that we use exactly the number of blocks its contents are from.
+			//
+			// For the last reply, we calculate the difference between the query's end_blocknum and
+			// the start of the reply.
+			// Overflow safe since end_blocknum must be >= msg.first_block_num.
 			let number_of_blocks = if batch_index == batch_count-1 {
-				msg.end_blocknum()
-			} else {
-				block_from_scid(batch.last().unwrap()) + 1
+				msg.end_blocknum() - first_blocknum
+			}
+			// Prior replies should use the number of blocks that fit into the reply. Overflow
+			// safe since first_blocknum is always <= last SCID's block.
```

Comment:
Hmm, for this to be accurate, wouldn't it be:
```
block_from_scid(batch.last()) - block_from_scid(batch.first()) + 1
```
? Because the comment seems to indicate that it's it the number of blocks in the current reply's batch?